### PR TITLE
[CPU]Fix profile graph

### PR DESCRIPTION
**Pull request by zhangYiIntel**
### Details:
 - *Fix profile graph construction*

### Tickets:
 - *CVS-136859*
 - https://github.com/openvinotoolkit/openvino.genai/issues/327

### CODE REVIEW

File: src/plugins/intel_cpu/src/graph_dumper.cpp [hunk 1]
```diff
@@ -164,8 +164,11 @@ std::shared_ptr<ov::Model> dump_graph_as_ie_ngraph_net(const Graph &graph) {
             results.emplace_back(std::make_shared<ov::op::v0::Result>(get_inputs(node).back()));
             return_node = results.back();
         } else {
+            // Even though outConfs.size() equals childEdges.size() during initailization,
+            // CPUNode's children node may be changed by graphOptimizer such as ShareReOrder
+            // The output_size here must use actual size of childEdges, otherwise out-of-index happens in getChildEdgeAt
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/graph_dumper.cpp:169]: If I read this comment correctly ShareReorder transformation breaks port configuration. I would say this is not expected behavior and should be fized inside the transformation itself.
@maxnick  @usstq guys, could you please comment?

Comment (zhangYiIntel) [reply src/plugins/intel_cpu/src/graph_dumper.cpp:169]: The problem here is that `outConfs` is configured inside  `InitOptimalPrimitiveDescriptors`, when `ShareReorders`([link](https://github.com/openvinotoolkit/openvino/blob/5c0d1468d53a1908a566b8a1bac41ae5dba72462/src/plugins/intel_cpu/src/graph.cpp#L248)) removes the edges from broadcast to duplicated Reorders, it only shrinks childEdges, there is no API to also configure the `outConfs`

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/graph_dumper.cpp:169]: Than such API should be added. 
Port descriptors and edges configuration should be consistent during whole model compilation/execution

**Comment by zhangYiIntel** [comment]
@usstq The `ShareReorders` optimizer is fixed with proper outConfs change. Please have a review

**Comment by zhangYiIntel** [comment]
@dmitry-gorokhov  It looks like that the `Broadcast` node confuses the concept between `Port` and `Edges`. Should `outConfs` be the same size with `outputShapes` ? From my review, the `Split` node uses `outputShapes` to initialize the size of `outConf`

File: src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp [hunk 2]
```diff
@@ -121,8 +121,9 @@ std::vector<NodeDesc> TileBroadcastCommon::getSupportedConfigs(const Node *node)
         config.inConfs[2].constant(constMap[2]);
         config.inConfs[2].setMemDesc(std::make_shared<CpuBlockedMemoryDesc>(ov::element::i32, node->getInputShapeAtPort(2)));
     }
-
-    config.outConfs.resize(node->getChildEdges().size());
+    //outConfs should be same with ports not edges, The structure is node->port->childEdge.
+    outSize = outSize ? outSize : node->getChildEdges().size();
+    config.outConfs.resize(outSize);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp:126]: Why cannot we just write the following: 
```
config.outConfs.resize(node->outputShapes.size());
```

Comment (zhangYiIntel) [reply src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp:126]: Sounds good, let me fix this

Comment (zhangyi93) [reply src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp:126]: The reason is that `outputShapes` is a protected member of `Node`. `TileBroadcastCommon` could not access it. We need to pass it as an argument of function `getSupportedConfigs` @dmitry-gorokhov 

Therefore, let me change this to the following directly
```
std::vector<NodeDesc> TileBroadcastCommon::getSupportedConfigs(const Node *node, size_t outSize) {
...
config.outConfs.resize(outSize);
...
```

Comment (zhangYiIntel) [reply src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp:126]: The reason is that outputShapes is a protected member of Node. TileBroadcastCommon could not access it. We need to pass it as an argument of function getSupportedConfigs @dmitry-gorokhov

Therefore, let me change this to the following directly
```
std::vector<NodeDesc> TileBroadcastCommon::getSupportedConfigs(const Node *node, size_t outSize) {
...
config.outConfs.resize(outSize);
...
```

File: src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp [hunk 3]
```diff
@@ -121,8 +121,8 @@ std::vector<NodeDesc> TileBroadcastCommon::getSupportedConfigs(const Node *node)
         config.inConfs[2].constant(constMap[2]);
         config.inConfs[2].setMemDesc(std::make_shared<CpuBlockedMemoryDesc>(ov::element::i32, node->getInputShapeAtPort(2)));
     }
-
-    config.outConfs.resize(node->getChildEdges().size());
+    //outConfs should be same with ports not edges, The graph structure is node->port->childEdge.
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/common/tile_broadcast_utils.cpp:124]: The comment itself is not need anymore. Lets remove it

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**